lib/medera/command_parser.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Medera.CommandParser do
+  @moduledoc """
+  Parses a command and resolves it against the registry of skills
+  """
+
+  alias Medera.Minion.Skill
+  alias Medera.Slack.Event
+
+  @type error_t :: :no_node | {:invalid_node, Skill.t}
+
+  @doc """
+  Parse the given text and resolve it against the given list of skills,
+  returning a matching skill on success.
+  """
+  @spec parse_command(binary, Event.t, [Skill.t]) ::
+  {:ok, Skill.t} | {:error, error_t}
+  def parse_command(text, _event = %Event{}, skills) do
+    [command | remainder] = String.split(text)
+    parse_command_arguments(Map.get(skills, command), remainder)
+  end
+
+  defp parse_command_arguments(nil, _), do: {:error, :no_match}
+  defp parse_command_arguments([skill], args) do
+    if skill.node_required do
+      case args do
+        [] -> {:error, :no_node}
+        [node | _other_args] ->
+          if Skill.valid_node?(skill, node) do
+            {:ok, skill}
+          else
+            {:error, {:invalid_node, skill}}
```

Comment:
It could be a node that is available but isn't a valid node for that skill.  For example `!show-postgres-processes minion@redishost` where `redishost` is a redis server and its minion only knows about redis things.